tedana/decomposition/gift_pca.py

```diff
@@ -0,0 +1,599 @@
+"""
+PCA based on GIFT software
+"""
+import logging
+
+import numpy as np
+from sklearn.decomposition import PCA, TruncatedSVD
+from sklearn.preprocessing import StandardScaler
+
+from scipy.signal import detrend
+from scipy.fftpack import fft, fftshift, fftn, fft2
+from scipy.signal import correlate2d
+
+LGR = logging.getLogger(__name__)
+
+
+def _autocorr(x):
+    """
+    Run Singular Value Decomposition (SVD) on input data,
+    automatically select components based on the GIFT software.
+
+    Parameters
+    ----------
+    x :
+
+    Returns
+    -------
+    u :
+    """
+    result = np.correlate(x, x, mode='full')
+    return result[result.size / 2:]
+
+
+def _sumN(dat):
+    """
+    Sum of all the elements of the dat matrix.
+
+    Parameters
+    ----------
+    dat :
+
+    Returns
+    -------
+    u :
+    """
+    return np.sum(dat[:])
+
+
+def _checkOrder(n_in):
+    """
+    Checks the order passed to the window functions.
+
+    Parameters
+    ----------
+    n_in :
+
+    Returns
+    -------
+    n_out :
+    w :
+    trivalwin :
+    """
+
+    w = []
+    trivalwin = 0
+
+    # Special case of negative orders:
+    if n_in < 0:
+        raise ValueError('Order cannot be less than zero.')
+
+    # Check if order is already an integer or empty
+    # If not, round to nearest integer.
+    if not n_in or n_in == np.floor(n_in):
```

Comment:
We should never hit `n_in=None`, here, since it's explicitly passed to the function and should have been sanitized previously (I assume, since the case isn't handled elegantly internally):

```suggestion
    if n_in == np.floor(n_in):
```

Reply:
Sure. It should never be `n_in=None`. I'll commit your suggestion.